dipy/core/tests/test_geometry.py

```diff
@@ -69,9 +69,9 @@ def test_sphere_cart():
     xyz = sphere2cart(r, theta, phi)
     yield assert_array_almost_equal, xyz, pt
 
-    # Test full circle on x=0, y=0, z=0
-    x, y, z = sphere2cart(*cart2sphere(0., 0., 0.))
-    yield assert_array_equal, (x, y, z), (0., 0., 0.)
+    # Test full circle on x=1, y=1, z=1
+    x, y, z = sphere2cart(*cart2sphere(1.0, 1.0, 1.0))
```

Comment:
Ah - this one doesn't raise an error. Instead, it raises an annoying warning (`RuntimeWarning: invalid value encountered in double_scalars theta = np.arccos(z / r)`), because `r` is set to 0. That actually  makes sense to me. Why would you have a zero-length vector? So, I think that we should let that warning come up, but I am not sure why we should test with a nonsense use-case.